### PR TITLE
[move-only] Ban @objc enum being marked as move only.

### PR DESCRIPTION
**Pull request by gottesmm**
rdar://106340608

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: lib/Sema/TypeCheckDeclPrimary.cpp [hunk 1]
```diff
@@ -2547,6 +2547,13 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       }
     }
 
+    // -----
+    // NonCopyableChecks
+    //
+
+    if (ED->isObjC() && ED->isMoveOnly()) {
+      ED->diagnose(diag::moveonly_objc_enum_banned);
```
Comment (slavapestov) [inline lib/Sema/TypeCheckDeclPrimary.cpp:2555]: Can you merge this with the checks on line 2560 below?

**Comment by gottesmm** [comment]
@swift-ci smoke test macOS platform

**Comment by gottesmm** [comment]
@swift-ci test windows platform

File: lib/Sema/TypeCheckDeclPrimary.cpp (hunk 1 above)
Comment (gottesmm) [reply lib/Sema/TypeCheckDeclPrimary.cpp:2555]: I am happy to do that in a subsequent commit. Can you be more specific... about what you want?

**Comment by gottesmm** [comment]
Forgot to add REQUIRES: objc_interop

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Pull request merged**